frontend/src/modules/app/OverlayScreen.tsx

```diff
@@ -36,34 +37,41 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
 }));
 
-export const BACKEND_CONNECTION_LOST_ALARM_TIMEOUT = 3000;
+export const BACKEND_CONNECTION_TIMEOUT = 3000;
 
 export const HeartbeatBackendListener = (): JSX.Element => {
   const clock = useSelector(getClock);
   const dispatch = useDispatch();
+  const heartbeat = useSelector(getBackendHeartBeat);
+  const events = useSelector(getNextLogEvents);
+  const diff = Math.abs(new Date().valueOf() - new Date(heartbeat).valueOf());
+
+  useEffect(() => {
+    if (diff < BACKEND_CONNECTION_TIMEOUT) {
+      dispatch(establishedBackendConnection(true, new Date()));
+    }
+  }, [clock, diff, dispatch, heartbeat]);
 
   useEffect(() => {
```

Comment:
Since these just cover different branches of the same `if` check (i.e. `diff <= BACKEND_CONNECTION_TIMEOUT`), would it make sense to merge these two into the same `useEffect`? Like:
```
if (diff <= BACKEND_CONNECTION_TIMEOUT {
  ...
} else if (!lostConnectionAlarm) {
  ...
}
```